server/server.go

```diff
@@ -293,15 +387,24 @@ func (s *RPCServer) responder(outCh *amqp.Channel) error {
 		)
 
 		if err != nil {
-			logger.Warnf("server: could not publish response, will retry later")
+			// An error here means that the connection that outCh is using
+			// isn't working properly. We trust that this function will be
+			// restarted by listenAndServe in this case.
+
+			// We resend the response here so that other running goroutines
+			// that hav a working outCh can pick up this response.
 			s.responses <- response
-			return err
 		}
+	}
+}
 
-		logger.Infof(
-			"server: successfully published response %v to %v",
-			response.publishing.CorrelationId,
-			response.replyTo,
-		)
+// Stop will gracefully disconnect from AMQP after draining first incoming then
+// outgoing messages. Use noWait if waiting for the exit isn't necessary.
+func (s *RPCServer) Stop(noWait bool) {
```

Comment:
I think it looks kinda bad when you see all the "anonymous" falses passed in every example and everywhere in the code.

So related to my question about the `stoppedChan`, what do you think about implementing `Stop()` with no argument which will always wait and then another `Kill()` which will never wait? I think that will make the code look better and also invite the usage of a graceful shutdown even more.

Reply:
Agreed